polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1546,6 +1546,8 @@ mlir::Type CodeGenTypes::getMLIRType(clang::QualType QT, bool *ImplicitRef,
       bool InnerSYCL = false;
       if (auto ST = dyn_cast<mlir::LLVM::LLVMStructType>(SubType))
         InnerSYCL |= any_of(ST.getBody(), mlir::sycl::isSYCLType);
+      else if (auto ST = dyn_cast<polygeist::StructType>(SubType))
+        InnerSYCL |= any_of(ST.getBody(), mlir::sycl::isSYCLType);
```

Comment:
Maybe using a `TypeSwitch` would be cleaner here, combining both `if` statement bodies

Reply:
done.